app/Models/SizeVariant.php

```diff
@@ -0,0 +1,243 @@
+	 */
+	protected $appends = [
+		'url',
+	];
+
+	/**
+	 * Returns the association to the photo which this size variant belongs
+	 * to.
+	 *
+	 * @return BelongsTo
+	 */
+	public function photo(): BelongsTo
+	{
+		return $this->belongsTo(Photo::class);
+	}
+
+	/**
+	 * Returns the association to the symbolics links which point to this
+	 * size variant.
+	 *
+	 * @return HasManyBidirectionally
+	 */
+	public function sym_links(): HasManyBidirectionally
+	{
+		return $this->hasManyBidirectionally(SymLink::class);
+	}
+
+	/**
+	 * Accessor for the "virtual" attribute {@link SizeVariant::$url}.
+	 *
+	 * This is more than a simple convenient method which wraps
+	 * {@link SizeVariant::$short_path} into
+	 * {@link \Illuminate\Support\Facades\Storage::url()}.
+	 * Based on the current application settings and the authenticated user,
+	 * this method returns a URL to a short-living symbolic link instead of a
+	 * direct URL to the actual size variant, if the underlying storage
+	 * provides symbolic links.
+	 *
+	 * @return string the url of the size variant
+	 */
+	public function getUrlAttribute(): string
+	{
+		if (
+			(AccessControl::is_admin() && Configs::get_value('SL_for_admin', '0') === '0') ||
+			Configs::get_value('SL_enable', '0') == '0'
+		) {
+			return Storage::url($this->short_path);
+		}
+
+		// In order to allow a grace period, we create a new symbolic link,
+		// if the most recent existing link has reached 2/3 of its lifetime
+		$maxLifetime = intval(Configs::get_value('SL_life_time_days', '3')) * 24 * 60 * 60;
+		$gracePeriod = $maxLifetime / 3;
+
+		$storageAdapter = Storage::disk()->getDriver()->getAdapter();
+
+		// TODO: Uncomment these line when Laravel really starts to support s3
+		/*if ($storageAdapter instanceof AwsS3Adapter) {
+			return Storage::temporaryUrl($this->short_path, now()->addSeconds($maxLifetime));
+		}*/
+
+		if ($storageAdapter instanceof Local) {
+			/** @var ?SymLink $symLink */
+			$symLink = $this->sym_links()->latest()->first();
+			if ($symLink == null || $symLink->created_at->isBefore(now()->subSeconds($gracePeriod))) {
+				/** @var SymLink $symLink */
+				$symLink = $this->sym_links()->create();
+			}
+
+			return $symLink->url;
+		}
+
+		throw new \InvalidArgumentException('the chosen storage adapter "' . Storage::getDefaultDriver() . '" does not support the symbolic linking feature');
+	}
+
+	/**
+	 * Accessor for the "virtual" attribute {@link SizeVariant::$full_path}.
+	 *
+	 * Returns the full path of the size variant as it needs to be input into
+	 * some low-level PHP functions like `unlink`.
+	 * This is a convenient method and wraps {@link SizeVariant::$short_path}
+	 * into {@link \Illuminate\Support\Facades\Storage::path()}.
+	 *
+	 * @return string the full path of the file
+	 */
+	public function getFullPathAttribute(): string
+	{
+		return Storage::path($this->short_path);
+	}
+
+	/**
+	 * Mutator of the attribute {@link SizeVariant::$size_variant}.
+	 *
```

Comment:
I would add in the comment that Original is 0 and Thumb is 7. :)

Reply:
See two lines below. The allowed values are documented in the PHPdoc directive `@param`.